config/critters-webpack-plugin.js

```diff
@@ -35,44 +34,54 @@ module.exports = class CrittersWebpackPlugin {
 
   /** Invoked by Webpack during plugin initialization */
   apply (compiler) {
-    const outputPath = compiler.options.output.path;
-
     // hook into the compiler to get a Compilation instance...
     compiler.hooks.compilation.tap(PLUGIN_NAME, compilation => {
       // ... which is how we get an "after" hook into html-webpack-plugin's HTML generation.
       compilation.hooks.htmlWebpackPluginAfterHtmlProcessing.tapAsync(PLUGIN_NAME, (htmlPluginData, callback) => {
-        // Parse the generated HTML in a DOM we can mutate
-        const document = parse5.parse(htmlPluginData.html, PARSE5_OPTS);
-        makeDomInteractive(document);
-
-        let externalStylesProcessed = Promise.resolve();
-
-        // `external:false` skips processing of external sheets
-        if (this.options.external !== false) {
-          const externalSheets = document.querySelectorAll('link[rel="stylesheet"]');
-          externalStylesProcessed = Promise.all(externalSheets.map(
-            link => this.embedLinkedStylesheet(link, compilation, outputPath)
-          ));
-        }
-
-        externalStylesProcessed
-          .then(() => {
-            // go through all the style tags in the document and reduce them to only critical CSS
-            const styles = document.querySelectorAll('style');
-            return Promise.all(styles.map(style => this.processStyle(style, document)));
-          })
-          .then(() => {
-            // serialize the document back to HTML and we're done
-            const html = parse5.serialize(document, PARSE5_OPTS);
-            callback(null, { html });
-          })
+        this.process(compiler, compilation, htmlPluginData)
+          .then(result => { callback(null, result); })
           .catch(callback);
       });
     });
   }
 
+  readFile (filename, encoding) {
+    return new Promise((resolve, reject) => {
+      this.fs.readFile(filename, encoding, (err, data) => {
```

Comment:
Where does `this.fs` come from? Why not use https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original?